llvm/lib/SYCLLowerIR/PrepareSYCLNativeCPU.cpp

```diff
@@ -276,107 +283,31 @@ PreservedAnalyses PrepareSYCLNativeCPUPass::run(Module &M,
   // Then we iterate over all the supported builtins, find their uses and
   // replace them with calls to our Native CPU functions.
   for (auto &Entry : BuiltinNamesMap) {
-    // Kernel -> builtin materialization CallInst, this is used to avoid
-    // inserting multiple calls to the same builtin
-    std::map<Function *, CallInst *> BuiltinCallMap;
-    // Map that associates to each User of a builtin, the index of the builtin
-    // in its operand list, and the callinst that will replace the builtin
-    std::map<User *, std::pair<unsigned, CallInst *>> ToReplace;
-    // We need to handle GEPOperator uses in a separate case since they are
-    // constants
-    std::set<GEPOperator *> GEPOps;
-    // spirv builtins are global constants, find it in the module
-    auto *Glob = M.getNamedGlobal(Entry.first);
+    auto *Glob = M.getFunction(Entry.first);
     if (!Glob)
       continue;
-    auto *ReplaceFunc = getReplaceFunc(M, StatePtrType, Entry.second);
+    auto *ReplaceFunc = getReplaceFunc(M, Entry.second.first);
+    SmallVector<Instruction *> ToRemove;
     for (auto &Use : Glob->uses()) {
```

Comment:
Could `Use` be const? I think it's not modified anywhere.

Reply:
Done, thanks